analyzers/src/SonarAnalyzer.CSharp.Styling/Rules/AvoidListForEach.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp.Styling;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class AvoidListForEach : StylingAnalyzer
+{
+    public AvoidListForEach() : base("T0011", "Use 'foreach' iteration instead of 'List.ForEach'.") { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+            {
+                var name = ((MemberAccessExpressionSyntax)c.Node).Name;
+                if (name.Identifier.ValueText == nameof(List<int>.ForEach)
+                    && c.SemanticModel.GetSymbolInfo(name).Symbol is IMethodSymbol method
+                    && method.Is(KnownType.System_Collections_Generic_List_T, "ForEach"))
```

Comment:
```suggestion
                    && method.Is(KnownType.System_Collections_Generic_List_T, nameof(List<int>.ForEach))
```

Reply:
I've added a few more tickets as I was implementing others - mainly by noticing stuff.

This one is not explicitly in our coding style, we generally agreed that `ForEach` is evil and we need to get rid of it and not use it in general.
There are few places in the codebse where this is used, and this rule will help get rid of them once we meet those rules.

Reply:
I've also added #9085 when working on the "is null", because that one is also easy to miss